### PR TITLE
Secrets manager Update.

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description

This PR adds minimum and complete examples to the module. It also adds all standard template files.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-secretsmanager/blob/feature/files-update/CHANGELOG.md#<VERSION_WITHOUT_DOTS>---<DATE>)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

**Comment by j0hnmwangi** [comment]
[checkov.txt](https://github.com/boldlink/terraform-aws-secretsmanager/files/8850388/checkov.txt)
Checkov logs

File: examples/binary/example.txt [hunk 1]
```diff
@@ -1 +1 @@
-ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDMl1fu1LkwUgHghZ137jXfHqgMrx+VVhADRQGF5ru0J8tP4NlSMB6B64+C9YoEhZm+gq8LTB6BbEAPyMiVYDEdGo2pKJv369c5BR0eLluLxQo37ckUTpUCnMsRd0UBxMv88Xl0tKYrgTjOZMUbX/+YqI+Bo/VyB1ghiiT31/qo820i6k9t/8/ktJEeQxcfinKgnyh0O/ds6s7+T2BOmx0CuIMIPMf43wOZB46IHGs/DOZQN+2xyJ1hv24XoxpbO/Lxq4XseqcsTzBlOqUPpkZhMfvWgloE80qjacu1aMd8nTB63ESlsGM3TPo7N4fxUIAe2PCKKIk1XgPNKc33lZ/D 
\ No newline at end of file
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDMl1fu1LkwUgHghZ137jXfHqgMrx+VVhADRQGF5ru0J8tP4NlSMB6B64+C9YoEhZm+gq8LTB6BbEAPyMiVYDEdGo2pKJv369c5BR0eLluLxQo37ckUTpUCnMsRd0UBxMv88Xl0tKYrgTjOZMUbX/+YqI+Bo/VyB1ghiiT31/qo820i6k9t/8/ktJEeQxcfinKgnyh0O/ds6s7+T2BOmx0CuIMIPMf43wOZB46IHGs/DOZQN+2xyJ1hv24XoxpbO/Lxq4XseqcsTzBlOqUPpkZhMfvWgloE80qjacu1aMd8nTB63ESlsGM3TPo7N4fxUIAe2PCKKIk1XgPNKc33lZ/D
```
Comment (hugoalmeida264) [inline examples/binary/example.txt]: This shouldn't be here it is better to use a terraform resource, did pre-commit failed to flag this? It should had picked it up.

File: examples/binary/locals.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+locals {
+  filename = "example.txt"
+  policy   = <<POLICY
```
Comment (hugoalmeida264) [inline examples/binary/locals.tf]: Use `jsoncode()`

File: examples/complete/locals.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+locals {
+  name     = "example-complete-secret"
+  filename = "secrets_manager_rotation.zip"
+  policy   = <<POLICY
```
Comment (hugoalmeida264) [inline examples/complete/locals.tf]: use `jsoncode()`

File: examples/key-pair/locals.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+locals {
+  policy = <<POLICY
```
Comment (hugoalmeida264) [inline examples/key-pair/locals.tf]: use data or `jsoncode()`

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]
- Add tags to the examples
 - Remove zip's and place them in code, there is a tf resource to package the code
 - Remove the ssh file, use a tf resource for it

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- Feature: Secrets manager
+- Feature: Secrets rotation
+- Feature: Secrets version
+- feature: Secrets policy
+- ...
+
+[Unreleased]: https://github.com/boldlink/terraform-aws-secretsmanager/compare/1.0.2...HEAD
+
+[1.0.2]: https://github.com/boldlink/terraform-aws-secretsmanager/releases/tag/1.0.2
+
```
Comment (hugoalmeida264) [inline CHANGELOG.md:32]: No line break

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+[1.0.1]: https://github.com/boldlink/terraform-aws-secretsmanager/releases/tag/v1.0.1
+
```
Comment (hugoalmeida264) [inline CHANGELOG.md:34]: No line break

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+
+## [Unreleased]
+- Fix: Test secret rotation with an rds instance
+## [1.0.2] - 2022-06-06
```
Comment (hugoalmeida264) [inline CHANGELOG.md:10]: Add line break

**Review by hugoalmeida264 — APPROVED** [review]

**Pull request merged**